### ISSUE TITLE
Compilation error in generated code when <enumConverter/> is applied to an enum column

### ISSUE THREAD

**Issue by Fadelis** [jOOQ/jOOQ#6930]
It would be great and would reduce unnecessary boilerplate code if forced enum types could be mapped without specifying a converter. No you have to create and converter class extending `EnumConverter` for each enum.

When specifying `forcedType`:
```xml
<forcedType>
  <userType>com.project.MyEnume</userType>
  <expressions>.*column_name.*</expressions>
</forcedType>
```
It would automatically map to and enum without the converter.

**Comment by lukaseder** [comment]
Indeed, this should already be possible. See:

- https://github.com/jOOQ/jOOQ/issues/5877
- https://github.com/jOOQ/jOOQ/issues/6567

This doesn't seem to have been documented in the manual yet. Will do soon: #6932

**Comment by Fadelis** [comment]
Would it be possible that this option `<enumConverter>true</enumConverter>` woudl work not only for test based columns, but also from SQL enum type columns?

**Comment by lukaseder** [comment]
@Fadelis You mean you want to translate the jOOQ-generated enum types to your own custom enum types? If so, what's the use-case?

**Comment by Fadelis** [comment]
For example, I have and enum type `DayOfWeek` created at database level, and at code level I want to use `java.time.DayOfWeek` for its API. Currently it works if I provide an `EnumConverter` in config, but if I use the `enumConverter` flag, during generation I get an error 
`org.jooq.exception.SQLDialectNotSupportedException: Type USER-DEFINED is not supported in dialect POSTGRES`

**Comment by lukaseder** [comment]
Oh, I see. That seems to be the same issue as this one here: https://stackoverflow.com/q/47749251/521799. Yeah, definitely not a duplicate, then. Will look into this soon.

**Comment by lukaseder** [comment]
Hmm, no this is a different issue than the stack overflow question. I can reproduce this. The generated code no longer compiles because the table is missing but it is referenced.

**Comment by lukaseder** [comment]
This is now fixed for jOOQ 3.11. The change was significant as some type information from the `GeneratorStrategy` (jOOQ-codegen) needed to be passed to the `DataTypeDefinition` (jOOQ-meta), thus I will not merge this fix to older versions. The workaround is to create manual `EnumConverters` on a case-by case basis.
